s30576-pj-NatanBrzozowski/zad_procesy_i_strumienie.txt

```diff
@@ -0,0 +1,239 @@
+     1886       1    1886      11540  ?         197609   Oct 30 /usr/bin/mintty
+     1988    1987    1988       9076  pty1      197609 22:42:37 /usr/bin/bash
+
+natan@DESKTOP-OP762KV MINGW64 ~/repo/s30576-NatanBrzozowski (master)
+$ ps x
+      PID    PPID    PGID     WINPID   TTY         UID    STIME COMMAND
+     1887    1886    1887      12184  pty0      197609   Oct 30 /usr/bin/bash
+     2109    1988    2109      15744  pty1      197609 23:00:08 /usr/bin/ps
+     1987       1    1987      16040  ?         197609 22:42:37 /usr/bin/mintty
+     1886       1    1886      11540  ?         197609   Oct 30 /usr/bin/mintty
+     1988    1987    1988       9076  pty1      197609 22:42:37 /usr/bin/bash
+
+natan@DESKTOP-OP762KV MINGW64 ~/repo/s30576-NatanBrzozowski (master)
+$ ps aux
+      PID    PPID    PGID     WINPID   TTY         UID    STIME COMMAND
+     1887    1886    1887      12184  pty0      197609   Oct 30 /usr/bin/bash
+     2114    1988    2114       9624  pty1      197609 23:00:13 /usr/bin/ps
+     1987       1    1987      16040  ?         197609 22:42:37 /usr/bin/mintty
+     1886       1    1886      11540  ?         197609   Oct 30 /usr/bin/mintty
+     1988    1987    1988       9076  pty1      197609 22:42:37 /usr/bin/bash
+
+natan@DESKTOP-OP762KV MINGW64 ~/repo/s30576-NatanBrzozowski (master)
+$
+
+--------------------------------------------------------------------------------
+|=============================|
+|         ZADANIE 2:          |
+|=============================|
+otworzylem gedit
+a nastepnie w drugim oknie terminala wyszukałem komenda ps aux | grep gedit
+i zamknalem go wykonujac kill 4454 - zamknal sie :D
+komenda killall bash wyslalem sygnal zakonczenia do wszystkich procesow 
+o nazwie "bash"
+
+
+
+|-----------------------------|
+|  FRAGMENT KODU Z TERMINALA: |
+|-----------------------------|
+natan@natan-virtual-machine:~$ ps aux | grep gedit
+natan       4454  4.5  0.3 559444 56580 pts/0    Sl+  23:49   0:00 gedit
+natan       4485  0.0  0.0  11740  2688 pts/1    S+   23:49   0:00 grep --color=auto gedit
+natan@natan-virtual-machine:~$ kill 4454
+natan@natan-virtual-machine:~$ 
+
+--------------------------------------------------------------------------------
+|=============================|
+|         ZADANIE 3:          |
+|=============================|
+rozumiem to tak ze w pierwszym przykladnie po wpisaniu witaj gdzy kliknalem 
+ctrl c skrot klawiszowy przerwał biezacy proces i nie zapisał tego co 
+wpisałem (witaj)
+a w drugim przykladnie gdzi wpisalem witaj i uzylem skrutu ctrl d biezacy proces
+zostal 
+poinformowany ze jest to koniec strumienia danych
+
+Ctrl+C:
+skrot klawiszowy ctrl c jest uzywany do przerwania lub zatrzymania biezacego 
+procesu w terminalu
+sluzy do natychmiastowego zakonczenia dzialania procesu
+
+Ctrl+D:
+skrot klawiszowy ctrl d jest uzywany do oznaczenia konca danych wejsciowych 
+lub sesji terminala
```

Comment:
Nie "lub sesji terminala". Sesja terminala się kończy bo nie ma więcej danych wejściowych. Jeśli inaczej zamkniesz strumień danych wejściowych to terminal także się zamknie. Nie trzeba do tego używać kombinacji `^D`.
Popraw proszę opis.